dafault.nix: GHC 8.8.3 as default; provide an easy part of the haskell.lib API

Anton-Latukha opened the pull request:
This would also make according to options in the CI work.

`doStrict` is ancient deprecated Nixpkgs option name, that is not used in Nixpkgs, nor in our `default.nix`, it is empty option.

For introduction `haskell.lib` options in CI before here. Sorry, that it is backward, I relearn the design flaws of Nix and its tooling usage again and again over the years.

These options provided - would start to work in CI now.

The rest of the options I would send-in PRs further.

---

Descriptions work, `nix-shell` works, `nix-build` works, project compiles, tests, benchmarks, profiles, with them enabled.

Anton-Latukha committed:
default.nix: make GHC 8.8.3 default

M  default.nix

Anton-Latukha committed:
default.nix: rm allowUnfree - there is none

M  default.nix

Anton-Latukha committed:
default.nix: add haskell.lib profiling options explicitly

Integrates with CI.

M  default.nix

Anton-Latukha committed:
default.nix: add flags {doCheck,doHaddock,doCoverage}

M  default.nix

sjakobi reviewed (dismissed):
I'll approve just so you can merge if you want, @Anton-Latukha. I haven't taken a closer look.

Thanks for all the work you do! :)

Anton-Latukha committed:
CI: Travis: build.sh: init CACHIX_SIGNING_KEY with default empty value

Due to new strict `set` rules `if` statement check agains empty variable was
failing the build, this fixes builds in PRs.

...sigh.

M  build.sh

Anton-Latukha committed:
default.nix: expose more features of `haskell.lib` API

M  default.nix

Anton-Latukha committed:
default.nix: add docs to the exposed haskell.lib options

M  default.nix
